tensorflow/python/framework/device_spec.py

```diff
@@ -333,17 +333,21 @@ def _string_to_components(spec=None):
           replica = y[1]
         elif ly == 2 and y[0] == "task":
           task = y[1]
-        elif ((ly == 1 or ly == 2) and (y[0].upper() in _VALID_DEVICE_TYPES)):
+        elif ((ly == 1 or ly == 2) and (y[0] != "")):
           if device_type is not None:
             raise ValueError("Cannot specify multiple device types: %s" % spec)
           device_type = y[0].upper()
           if ly == 2 and y[1] != "*":
+            if y[1].isdigit() is False:
+              raise ValueError("invalid literal for int: %s" % spec)
```

Comment:
Nit:
```suggestion
              raise ValueError("Invalid literal for int: %s" % spec)
```
Same for the other place below.

Reply:
I see. This is what `int()` throws when its input is invalid. I don't think we need to add explicit checks here and in lines 349-350.